mlflow/models/evaluation/base.py

```diff
@@ -1140,7 +1173,9 @@ def _convert_data_to_mlflow_dataset(data, targets=None, predictions=None):
         from pyspark.sql import DataFrame as SparkDataFrame
 
     if isinstance(data, list):
-        return mlflow.data.from_numpy(np.array(data), targets=np.array(targets))
+        return mlflow.data.from_numpy(
+            np.array(data), targets=np.array(targets) if targets else None
```

Comment:
note: This is small but important change. In the original logic, ``targets`` param is always wrapped by numpy array when the input data is list, then ``if targets is None`` doesn't work in downstream. Concretely, when we pass a list as an input data and leave targets ``None``,  it will raise a weird error. 

This is not specific to the endpoint evaluation, however, implicitly requiring targets only for list input is confusing, so I think we should fix it in this PR.